### PR TITLE
`Quiz exercises`: Add single choice mode for multiple choice questions

### LINKED ISSUE #3635: Add option for single choice questions in quiz exercises
### Is your feature request related to a problem?

Some quiz questions only expect one answer, which is also stated more or less clearly in the problem statement of that question.
But Artemis still displays "Please choose all correct answer options" above the answer options and allows the student to select multiple answers. This can lead to some confusion among students.

### Describe the solution you'd like

There should be an option for the creator of the quiz to mark a question as single choice. This should change the displayed text above the answer options, to signal the student clearly that only one option is correct, and only allow one checked answer option.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by bottbenj**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Server
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/server/).
- [x] I added multiple integration tests (Spring) related to the features (with a high test coverage).
- [ ] N/A. ~~I added `@PreAuthorize` and checked the course groups for all new REST Calls (security).~~
- [x] I implemented the changes with a good performance and prevented too many database calls.
- [x] I documented the Java code using JavaDoc style.
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-tests/).
- [ ] N/A ~~I added `authorities` to all new routes and checked the course groups for displaying navigation elements (links, buttons).~~
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context
Closes #3635

### Description
Adds a new option `Single Choice Mode` to multiple choice questions. If enabled it enforces that the question has exactly one correct answer and scoring mode `ALL_OR_NOTHING`.

From the student view it changes the text above the answer options to indicate that only one answer is correct and makes the answer options behave like radio buttons, in that at most one can be selected at a time. It is still possible to select no options to support possible future scoring modes where not answering may be desirable over guessing. Furthermore the scoring info dialogue is adjusted slightly as these questions cannot be answered partially correctly.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student
- 1 Quiz Exercise with multiple Choice Questions

1. Edit the quiz
2. Play around with the single choice option, the scoring mode and different combinations of correct and wrong answer options
3. Check that the scoring mode always is `ALL_OR_NOTHING` if single choice is activated and that that quiz can only be saved if there is exactly one correct answer for single choice questions.
4. Start & Participate in the Quiz
5. Ensure that it is clearly visible which questions are single choice and which are not. Ensure that for single choice questions at most one answer option can be selected at a time.

#### Code Review
- [x] Review 1
- [ ] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Screenshots

Edit MC Questions:
image

Answer Single Choice Mode Question:
image

### CODE REVIEW

File: src/main/webapp/app/exercises/quiz/manage/quiz-exercise-validation.directive.ts [hunk 1]
```diff
@@ -84,18 +83,18 @@ export abstract class QuizExerciseValidationDirective {
...
+                        question.title &&
+                        question.title !== '' &&
+                        question.title.length < this.maxLengthThreshold &&
+                        mcQuestion.answerOptions!.every(
+                            (answerOption) =>
+                                (!answerOption.explanation || answerOption.explanation.length <= this.explanationLengthThreshold) &&
+                                (!answerOption.hint || answerOption.hint.length <= this.hintLengthThreshold),
+                        )
+                    );
                     break;
```
Comment (Hialus) [inline src/main/webapp/app/exercises/quiz/manage/quiz-exercise-validation.directive.ts:98]: The break looks unnecessary to me now, as you always return now.

File: src/main/webapp/app/exercises/quiz/manage/quiz-exercise-validation.directive.ts [hunk 2]
```diff
@@ -1,19 +1,18 @@
 import { ChangeDetectorRef, Directive } from '@angular/core';
 import { QuizExercise } from 'app/entities/quiz/quiz-exercise.model';
 import dayjs from 'dayjs/esm';
-import { QuizQuestion, QuizQuestionType } from 'app/entities/quiz/quiz-question.model';
+import { QuizQuestion, QuizQuestionType, ScoringType } from 'app/entities/quiz/quiz-question.model';
```
Comment (willergeorg) [inline src/main/webapp/app/exercises/quiz/manage/quiz-exercise-validation.directive.ts:4]: There seems to be an unused import you can omit

File: src/main/webapp/app/exercises/quiz/manage/quiz-scoring-info-modal/quiz-scoring-info-modal.component.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit, Input } from '@angular/core';
```
Comment (willergeorg) [inline src/main/webapp/app/exercises/quiz/manage/quiz-scoring-info-modal/quiz-scoring-info-modal.component.ts:1]: Input seems to be unused as well

File: src/main/webapp/app/exercises/quiz/shared/questions/multiple-choice-question/multiple-choice-question.component.ts [hunk 4]
```diff
@@ -2,11 +2,11 @@ import { Component, EventEmitter, Input, Output, ViewEncapsulation } from '@angu
 import { ArtemisMarkdownService } from 'app/shared/markdown.service';
 import { AnswerOption } from 'app/entities/quiz/answer-option.model';
 import { MultipleChoiceQuestion } from 'app/entities/quiz/multiple-choice-question.model';
-import { RenderedQuizQuestionMarkDownElement } from 'app/entities/quiz/quiz-question.model';
+import { RenderedQuizQuestionMarkDownElement, ScoringType } from 'app/entities/quiz/quiz-question.model';
```
Comment (willergeorg) [inline src/main/webapp/app/exercises/quiz/shared/questions/multiple-choice-question/multiple-choice-question.component.ts:5]: This import is unused as well

File: src/test/javascript/spec/component/quiz-exercise/quiz-exercise-detail.component.spec.ts [hunk 5]
```diff
@@ -16,7 +16,7 @@ import { DragItem } from 'app/entities/quiz/drag-item.model';
 import { DropLocation } from 'app/entities/quiz/drop-location.model';
 import { MultipleChoiceQuestion } from 'app/entities/quiz/multiple-choice-question.model';
 import { QuizExercise } from 'app/entities/quiz/quiz-exercise.model';
-import { QuizQuestion, QuizQuestionType } from 'app/entities/quiz/quiz-question.model';
+import { QuizQuestion, QuizQuestionType, ScoringType } from 'app/entities/quiz/quiz-question.model';
```
Comment (willergeorg) [inline src/test/javascript/spec/component/quiz-exercise/quiz-exercise-detail.component.spec.ts:19]: Unused as well

**Review by willergeorg — DISMISSED** [review]
Tested your changes on TS1. Worked as expected for me and I did not find any undesired behavior. In the code, there seem to be some unused imports. I marked you the corresponding spots. Please remove them if they are not needed.

**Review by willergeorg — DISMISSED** [review]
Code looks fine now as well. 👍

**Review by jschmautzer — APPROVED** [review]
tested on ts3, followed all testing steps, worked as expected

File: src/main/webapp/i18n/de/quizExercise.json [hunk 6]
```diff
@@ -165,6 +165,7 @@
                 "questionTitle": "Frage {{index}}: Der Titel fehlt.",
                 "questionScore": "Frage {{index}}: Mindestens einen Punkt pro Aufgabe erforderlich.",
                 "questionCorrectAnswerOption": "Frage {{index}}: Es existiert keine korrekte Antwortmöglichkeit.",
+                "questionSingleChoiceCorrectAnswerOptions": "Frage {{index}}: Es existieren mehrere korrekte Antwortmöglichkeiten.",
```
Comment (JohannesStoehr) [inline src/main/webapp/i18n/de/quizExercise.json:168]:
```suggestion
                "questionSingleChoiceCorrectAnswerOptions": "Frage {{index}}: Es existieren mehrere korrekte Antwortmöglichkeiten bei einer Frage mit Einfachauswahl.",
```

File: src/main/webapp/i18n/en/quizExercise.json [hunk 7]
```diff
@@ -165,6 +165,7 @@
                 "questionTitle": "Question {{index}}: The title is missing.",
                 "questionScore": "Question {{index}}: At least one point is required per question.",
                 "questionCorrectAnswerOption": "Question {{index}}: No answer option is correct.",
+                "questionSingleChoiceCorrectAnswerOptions": "Question {{index}}: Multiple answer options are correct.",
```
Comment (JohannesStoehr) [inline src/main/webapp/i18n/en/quizExercise.json:168]:
```suggestion
                "questionSingleChoiceCorrectAnswerOptions": "Question {{index}}: Multiple answer options are correct for a single choice question.",
```

**Review by JohannesStoehr — APPROVED** [review]
Tested on TS1, only one small nitpick for the error message when multiple correct answers are added to a single choice question:

**Review by jschmautzer — APPROVED** [review]
re-approve, tested on ts3, still works as expected :)

**Review by xlieA — APPROVED** [review]
I tested on ts3 and followed the testing steps. Everything worked perfectly.

**Review by nairaarnez — APPROVED** [review]
Tested on ts3, works as espected

**Pull request merged**